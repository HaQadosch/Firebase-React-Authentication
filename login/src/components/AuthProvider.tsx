import React, { createContext, useState, useEffect } from 'react';
import { authApp } from '../base';

export const AuthContext = createContext({ currentUser: null });

export const AuthProvider: React.FC = ({ children }) => {
  const [currentUser, setCurrentUser] = useState();

  useEffect(() => {
    authApp.auth().onAuthStateChanged(setCurrentUser);
  }, []);

  return <AuthContext.Provider value={{ currentUser }}>{children}</AuthContext.Provider>;
};

export default AuthProvider;
