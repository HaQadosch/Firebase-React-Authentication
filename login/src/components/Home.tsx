import React from 'react';
import { authApp } from '../base';

export const Home = () => {
  return (
    <main>
      <h1>Home</h1>
      <button onClick={() => authApp.auth().signOut()}>Sign Out</button>
    </main>
  );
};

export default Home;
