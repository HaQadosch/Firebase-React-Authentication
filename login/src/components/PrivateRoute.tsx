import React, { useContext } from 'react';
import { Route, Redirect } from 'react-router-dom';
import { AuthContext } from './AuthProvider';

interface MyProps {
  component: any;
  exact: boolean;
  path: string;
}

export const PrivateRoute: React.FC<MyProps> = ({ component: RouteComponent, ...rest }) => {
  const { currentUser } = useContext(AuthContext);

  return (
    <Route render={routeProps => (!!currentUser ? <RouteComponent {...routeProps} /> : <Redirect to={'/login'} />)} />
  );
};

export default PrivateRoute;
