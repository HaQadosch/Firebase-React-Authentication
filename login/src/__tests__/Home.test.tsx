import React from 'react';
import ReactDOM from 'react-dom';
import { render } from 'react-testing-library';
import { Home } from '../components/Home';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Home />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('matches the snapshot', () => {
  const { container, debug } = render(<Home />);
  debug(container);
  expect(container.firstChild).toMatchSnapshot();
});
