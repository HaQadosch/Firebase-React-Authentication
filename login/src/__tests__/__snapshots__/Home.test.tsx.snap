// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<main>
  <h1>
    Home
  </h1>
  <button>
    Sign Out
  </button>
</main>
`;
